x-pack/plugins/observability/public/components/shared/exploratory_view/configurations/lens_attributes.test.ts

```diff
@@ -339,135 +344,7 @@ describe('Lens Attribute', () => {
-            references: ['y-axis-column-layer0X4'],
-            scale: 'ratio',
-          },
-          'y-axis-column-layer0X0': {
-            customLabel: true,
-            dataType: 'number',
-            filter: {
-              language: 'kuery',
-              query:
-                'transaction.type: page-load and processor.event: transaction and transaction.type : *',
-            },
-            isBucketed: false,
-            label: 'Part of count() / overall_sum(count())',
-            operationType: 'count',
-            scale: 'ratio',
-            sourceField: RECORDS_FIELD,
-          },
-          'y-axis-column-layer0X1': {
-            customLabel: true,
-            dataType: 'number',
-            filter: {
-              language: 'kuery',
-              query:
-                'transaction.type: page-load and processor.event: transaction and transaction.type : *',
-            },
-            isBucketed: false,
-            label: 'Part of count() / overall_sum(count())',
-            operationType: 'count',
-            scale: 'ratio',
-            sourceField: RECORDS_FIELD,
-          },
-          'y-axis-column-layer0X2': {
-            customLabel: true,
-            dataType: 'number',
-            isBucketed: false,
-            label: 'Part of count() / overall_sum(count())',
-            operationType: 'math',
-            params: {
-              tinymathAst: 'y-axis-column-layer0X1',
-            },
-            references: ['y-axis-column-layer0X1'],
-            scale: 'ratio',
-          },
-          'y-axis-column-layer0X3': {
-            customLabel: true,
-            dataType: 'number',
-            isBucketed: false,
-            label: 'Part of count() / overall_sum(count())',
-            operationType: 'overall_sum',
-            references: ['y-axis-column-layer0X2'],
-            scale: 'ratio',
-          },
-          'y-axis-column-layer0X4': {
-            customLabel: true,
-            dataType: 'number',
-            isBucketed: false,
-            label: 'Part of count() / overall_sum(count())',
-            operationType: 'math',
-            params: {
-              tinymathAst: {
-                args: ['y-axis-column-layer0X0', 'y-axis-column-layer0X3'],
-                location: {
-                  max: 30,
-                  min: 0,
-                },
-                name: 'divide',
-                text: "count(kql='transaction.type: page-load and processor.event: transaction and transaction.type : *') / overall_sum(count(kql='transaction.type: page-load and processor.event: transaction and transaction.type : *'))",
-                type: 'function',
-              },
-            },
-            references: ['y-axis-column-layer0X0', 'y-axis-column-layer0X3'],
-            scale: 'ratio',
-          },
-        },
-        incompleteColumns: {},
-      },
-    });
+    expect(lnsAttr.getLayers()).toEqual(sampleAttribute.state.datasourceStates.indexpattern.layers);
```

Comment:
I like to use `toMatchSnapshot` to compare huge objects like this one. I find it much easier to manage and maintain.

Reply:
snapshot are useful, but makes it hard to enforce, for this purpose i think json make sense, since it enforces you to review every single change.